src/d/actor/d_a_tpota.cpp

```diff
@@ -5,71 +5,171 @@
+    
+    unknown_struct* unknown_struct = field_0x2C4;
+    int ret = 0;
+    if (i_position_y <= -230.0f) {
+        for (int i = 0; i < ARRAY_SIZE(field_0x2C4); i++, unknown_struct++) {
+            if (unknown_struct->ptcl == i_ptcl) {
+                if (unknown_struct->pos_y > -230.0f) {
+                    ret = 1;
+                }
+                break;
+            }
+        }
+    }
+    return ret;
 }
 
 /* 00000354-00000380       .text clear_splash__9daTpota_cFv */
 void daTpota_c::clear_splash() {
-    /* Nonmatching */
+    unknown_struct* unknown_struct = field_0x2C4;
+    int count = 30;
+
+    while (count != 0) {
+        unknown_struct->ptcl = NULL;
+        unknown_struct->pos_y = 0.0;
+        unknown_struct++;
+        count--;
+    }
+    return;
 }
 
 /* 00000380-000003F4       .text renew_splash__9daTpota_cFv */
 void daTpota_c::renew_splash() {
-    /* Nonmatching */
+    
+
+    if (emitters[1] != NULL) {
+        JSUPtrList* list = &emitters[1]->mActiveParticles;
+        unknown_struct *unknown_struct = field_0x2C4;
+        if(list != NULL){
+            clear_splash();
+            for (JSUPtrLink* link = list->getFirstLink(); link != NULL && (link != NULL); link = link->getNext()){
+                JPABaseParticle* particle = (JPABaseParticle*)link->getObjectPtr();
+                JGeometry::TVec3<f32> position;
+                particle->getGlobalPosition(position);
+                unknown_struct->ptcl = particle;
+                unknown_struct->pos_y = position.y;
+                unknown_struct++;
+            }   
+            
+        }
+
+    }
+    return;
 }
 
 /* 000003F4-000004C8       .text _execute__9daTpota_cFv */
 bool daTpota_c::_execute() {
-    /* Nonmatching */
+    float local_38;
+    if (emitters[1] != NULL){
+        JSUPtrList* list = &emitters[1]->mActiveParticles;
+        if(list != NULL){
+            for (JSUPtrLink* link = list->getFirstLink(); link != NULL && (link != NULL);
+            link = link->getNext()){
+                JPABaseParticle* particle = (JPABaseParticle *)link->getObjectPtr();
+                JGeometry::TVec3<f32> position;
+                particle->getGlobalPosition(position);
+                int idk = check_water_h(particle, position.y);
+                if(idk != 0){
+                    cXyz local_48;
+                    local_48.x = position.x;
+                    local_48.y = -230.0;
+                    local_48.z = position.z;
+                    make_ripple(local_48);
+                }
+            }
+        }
+        renew_splash();
+    }
+    return 1;
 }
 
+
 /* 000004C8-000004D0       .text _draw__9daTpota_cFv */
 bool daTpota_c::_draw() {
-    /* Nonmatching */
+    return TRUE;
 }
 
 namespace {
 /* 000004D0-000004F0       .text Mthd_Create__23@unnamed@d_a_tpota_cpp@FPv */
-cPhs_State Mthd_Create(void*) {
-    /* Nonmatching */
+s32 Mthd_Create(void* i_this) {
```

Comment:
```suggestion
cPhs_State Mthd_Create(void* i_this) {
```
Looks like this got reverted by accident in a merge